PhysicsTools/PatAlgos/python/tools/jetTools.py

```diff
@@ -597,10 +597,18 @@ def setupBTagging(process, jetSource, pfCandidates, explicitJTA, pvSource, svSou
                                     btag.pixelClusterTagInfos.clone(jets = jetSource, vertices=pvSource),
                                     process, task)
 
-            if 'Puppi' in jetSource.value() and pfCandidates.value() == 'particleFlow' and\
-	       ('pfBoostedDouble' in btagInfo or 'SecondaryVertex' in btagInfo):
-                _btagInfo = getattr(process, btagPrefix+btagInfo+labelName+postfix)
-                _btagInfo.weights = cms.InputTag("puppi")
+            if 'pfBoostedDouble' in btagInfo or 'SecondaryVertex' in btagInfo:
+              _btagInfo = getattr(process, btagPrefix+btagInfo+labelName+postfix)
+              if pfCandidates.value() == 'packedPFCandidates':
+                _btagInfo.weights = "packedpuppi"
+                if not hasattr(process,"packedpuppi"):
+                  from CommonTools.PileupAlgos.Puppi_cff import puppi
+                  addToProcessAndTask('packedpuppi', puppi.clone(
+                        useExistingWeights = True,
+                        candName = 'packedPFCandidates',
+                        vertexName = 'offlineSlimmedPrimaryVertices') , process, task)
+              else:
+                _btagInfo.weights = "puppi"
```

Comment:
apparently here ```weights``` does not already exist.
Is this some different module type or was it not created from the fillDescriptions file?

Reply:
yes, there are many SecondaryVertex config files not created from fillDescriptions:
https://github.com/ahinzmann/cmssw/tree/master/RecoBTag/SecondaryVertex/python

Reply:
the weight parameter is however protected against misspelling, if it misses the code in this PR will throw a error.